pool/paymentmgr_test.go

```diff
@@ -947,4 +993,709 @@ func testPaymentMgr(t *testing.T, db *bolt.DB) {
+	pmtB := NewPayment(yID, randSource, amt, height, estMaturity)
+	mPmts[randSource.Coinbase] = []*Payment{pmtB}
+
+	ctx, cancel := context.WithCancel(context.Background())
+
+	// Ensure orphaned payments pruning returns an error if it cannot
+	// confirm a block.
+	mgr.cfg.GetBlockConfirmations = func(ctx context.Context, bh *chainhash.Hash) (int64, error) {
+		return 0, fmt.Errorf("unable to confirm block")
+	}
+	_, err = mgr.pruneOrphanedPayments(ctx, mPmts)
+	if err == nil {
+		cancel()
+		t.Fatal("expected a block confirmation error")
+	}
+
+	// Create an invalid block hash / payment set entry.
+	invalidBlockHash := "0123456789012345678901234567890123456789" +
+		"0123456789012345678912345"
+	mPmts[invalidBlockHash] = []*Payment{pmtB}
+
+	mgr.cfg.GetBlockConfirmations = func(ctx context.Context, bh *chainhash.Hash) (int64, error) {
+		if bh.String() != zeroSource.BlockHash {
+			return -1, nil
+		}
+		return 16, nil
+	}
+
+	// Ensure orphaned payments pruning returns an errors if it encounters
+	// an invalid block hash as a key.
+	_, err = mgr.pruneOrphanedPayments(ctx, mPmts)
+	if !errors.Is(err, ErrCreateHash) {
+		cancel()
+		t.Fatalf("expected a hash error, got %v", err)
+	}
+
+	// remove the invalid block hash key pair.
+	delete(mPmts, invalidBlockHash)
+
+	// Ensure orphaned payments pruning accurately prunes payments
+	// sourcing from orphaned blocks.
+	pmtSet, err = mgr.pruneOrphanedPayments(ctx, mPmts)
+	if err != nil {
+		cancel()
+		t.Fatalf("unexpected pruneOrphanPayments error: %v", err)
+	}
+	if len(pmtSet) != 1 {
+		cancel()
+		t.Fatalf("expected a single valid mature payment after "+
+			"pruning, got %v", len(pmtSet))
+	}
+
+	// applyTxFee tests.
+	tIn, _ := dcrutil.NewAmount(100)
+	in := chainjson.TransactionInput{
+		Amount: float64(tIn),
+		Txid:   chainhash.Hash{1}.String(),
+		Vout:   2,
+		Tree:   wire.TxTreeRegular,
+	}
+
+	poolFeeValue := amt.MulF64(0.1)
+	xValue := amt.MulF64(0.6)
+	yValue := amt.MulF64(0.3)
+
+	feeAddr := poolFeeAddrs.String()
+	out := make(map[string]dcrutil.Amount)
+	out[xAddr] = xValue
+	out[yAddr] = yValue
+	out[feeAddr] = poolFeeValue
+
+	tOut := tIn
+
+	_, txFee, err := mgr.applyTxFees([]chainjson.TransactionInput{in},
+		out, tOut, poolFeeAddrs)
+	if err != nil {
+		t.Fatalf("unexpected applyTxFees error: %v", err)
+	}
+
+	// Ensure the pool fee payment was exempted from tx fee deductions.
+	if out[feeAddr] != amt.MulF64(0.1) {
+		t.Fatalf("expected pool fee payment to be %v, got %v",
+			txFee, out[feeAddr])
+	}
+
+	// Ensure the difference between initial account payments and updated
+	// account payments plus the transaction fee is not more than the
+	// maximum rounding difference.
+	initialAccountPayments := xValue + yValue
+	updatedAccountPaymentsPlusTxFee := out[xAddr] + out[yAddr] + txFee
+	if initialAccountPayments-updatedAccountPaymentsPlusTxFee <= maxRoundingDiff {
+		t.Fatalf("initial account payment total %v to be equal to updated "+
```

Comment:
`cancel()`?

Reply:
No, `applyTxFees` is not using the context, therefore does not need the context cancelled for a clean shutdown.